__tests__/index.test.js

```diff
@@ -194,3 +194,51 @@ describe('GET /api/articles',()=>{
 });
 
 
+describe('GET /apri/articles (topic query',()=>{
+    test('return a 200 code when a request is made to the api',()=>{
+        return request(app)
+        .get('/api/articles?topic=mitch')
+        .expect(200)
+    })
+    test('return a 200 code and allarticles filtered by query when a valid request is made to the api "mitch"',()=>{
+        return request(app)
+        .get('/api/articles?topic=mitch')
+        .expect(200)
+        .then((response)=>{
+            expect(response.body.articles.length).toBe(12)
+        })
+    })
+    test('return a 200 code and all articles filtered by query when a valid request is made to the api "cats"',()=>{
+        return request(app)
+        .get('/api/articles?topic=cats')
+        .expect(200)
+        .then((response)=>{
+            expect(response.body.articles.length).toBe(1)
+        })
+    })
+    test('return a 200 code and all articles filtered by query when a valid request is made to the api "paper"',()=>{
+        return request(app)
+        .get('/api/articles?topic=paper')
+        .expect(200)
+        .then((response)=>{
+            expect(response.body.articles.length).toBe(0)
+        })
+    })
+    test('return a 404 code and msg when passed a valid by non-existant query',()=>{
+        return request(app)
+        .get('/api/articles?topic=trees')
+        .expect(404)
+        .then((response)=>{
+            expect(response.body.msg).toBe('No topic with that id')
+        })
+    })
+    test('return a 200 code when passed an empty query, eg. query would be looking for an topic= empty string',()=>{
```

Comment:
Should this be a valid 200 request in this scenario? Essentially we are testing for a topic not found here which we've already covered above.

Reply:
test removed